http/src/main/java/io/micronaut/http/originatingips/IpAddressesResolverAggregator.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.originatingips;
+
+import io.micronaut.http.HttpRequest;
+
+import javax.inject.Singleton;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Aggregator of beans of type {@link IpAddressesResolver}.
+ *
+ * @author Sergio del Amo
+ * @since 1.2.0
+ */
+@Singleton
+public class IpAddressesResolverAggregator {
```

Comment:
This should instead implement IpAddressesResolver and be marked as primary. I would also call it `CompositeIpAddressesResolver`

Reply:
I annotated it with `@Primary` and it implements `RequestIpAddressesResolver` interface.